kimchi/src/lagrange_basis_evaluations.rs

```diff
@@ -2,18 +2,85 @@ use ark_ff::{batch_inversion_and_mul, FftField};
+/// polynomial degree allowed. In particular, the SRS is not long enough to
+/// support higher degrees. Thus, we cannot obtain a polynomial f with degree
+/// c·n (with domain size n and number of chunks c) as usual with the equation:
+///
+/// f(X) = x_0 · l_0(X) + ... + x_n · l_n(X) + ... + x_{c*n} · l_{c*n}(X)
+///
+/// Instead, this struct will contain the (c*n+1) coefficients of the polynomial
+/// that is equal to the powers of the point x in the positions corresponding to
+/// the chunk, and 0 elsewhere in the domain. This is useful to evaluate the
+/// chunks of polynomials of degree c·n given in evaluation form at the point.
+///
 pub struct LagrangeBasisEvaluations<F> {
+    /// If no chunking:
+    /// - evals is a vector of length 1 containing a vector of size n
+    ///   corresponding to the evaluations of the Lagrange polynomials, which
+    ///   are the polynomials that equal 1 at ω_i and 0 elsewhere in the domain.
+    /// If chunking (a vector of size c * N )
+    /// - the first index refers to the chunks
+    /// - the second index refers j-th coefficient of the i-th chunk of the
+    ///   polynomial that equals the powers of the point and 0 elsewhere.
+    ///
     evals: Vec<Vec<F>>,
 }
 
 impl<F: FftField> LagrangeBasisEvaluations<F> {
-    /// Given the evaluations form of a polynomial, directly evaluate that polynomial at a point.
+    /// Given the evaluations form of a polynomial, directly evaluate that
+    /// polynomial at a point.
+    ///
+    /// The Lagrange basis evaluations can be used to evaluate a polynomial of
+    /// given in evaluation form efficiently in O(n) time, of degree d <= n
+    /// where n is the domain size, without the need of interpolating it first.
+    ///
+    /// Recall that a polynomial can be represented as the sum of the scaled
+    /// Lagrange polynomials using its evaluations on the domain:
+    /// $ f(x) = x_0 · l_0(x) + ... + x_n · l_n(x) $
+    ///
+    /// Recall that when chunking is in place, we want to evaluate a
+    /// polynomial f of degree c · m at point z, expressed as
+    ///
+    /// f(z) = a_0·z^0 + ... + a_{n-1}·z^{n-1} + ... + a_{c*n-1}·z^{c*n-1}
+    ///      = z^0 · f_0(z) + z^n · f_1(z) + ... + z^{(c-1)n} · f_{c-1}(z)
+    ///
+    /// where f_i(X) is the i-th chunked polynomial of degree m of f:
+    /// f_i(x) = a_{i*n} · x^0 + ... + a_{(i+1)n-1} · x^{n-1}
+    ///
+    /// Returns the evaluation of each chunk of the polynomial at the point
+    /// (when there is no chunking, the result is a vector of length 1). They
+    /// correspond to the f_i(z) in the equation above.
+    ///
     pub fn evaluate<D: EvaluationDomain<F>>(&self, p: &Evaluations<F, D>) -> Vec<F> {
+        // The domain size must be a multiple of the number of evaluations so
+        // that the degree of the polynomial can be split into chunks of equal size.
         assert_eq!(p.evals.len() % self.evals[0].len(), 0);
```

Comment:
It would be nice to have an alias for `self.evals[0].len()` as `self.domain_size()`.
We should also have tests enforcing that each way of constructing a structure of type `LagrangeBasisEvaluations` respect some invariants, like every chunks is of the same size (and therefore, using `self.evals[0].len()` or `self.evals[1].len()` is `self.evals.len() > 1` return the same output).